CMakeLists.txt

```diff
@@ -79,6 +79,40 @@ ELSE()
   OPTION(WITH_UPDATER "Include libraries and code to allow release versions to be updated on demand" ON)
 ENDIF()
 
+# Enable the inclusion of fonts currently carried in the source code by default
+# unless the environmental variable WITH_FONTS is already defined and is not
+# set to (case insenstive) "no". Linux packagers will find it useful to do this
+# since package managers may already package the relevant fonts - or are not
+# willing or able to include them.
+# Note: WITH_FONTS is an environmental value/variable (could be a number, a
+#           string, something else or not even exisit).
+#       FONT_TEST and USE_FONTS are CMake variables (probably a string and a
+#           boolean {ON or OFF} respectively).
+#       INCLUDE_FONTS is a C preprocessor symbol (i.e. the same as
+#           "#define INCLUDE_FONTS" which, in the absence of an explicitly set
+#           value, takes the numeric value of (int)1 as far as the compiler is
+#           concerned!)
+set(FONT_TEST $ENV{WITH_FONTS})
+if(DEFINED FONT_TEST)
+    string(TOUPPER ${FONT_TEST} FONT_TEST)
+else()
+    # Set the CMake variable to the default if it (as a result of WITH_FONTS not
+    # existing) has not been defined, without the if(DEFINED ...) string(...)
+    # errors out - and so may further tests on the variable...
+    set(FONT_TEST "YES")
+endif()
+if(FONT_TEST STREQUAL "NO")
+  option(USE_FONTS "Include Deja Vu and Ubuntu fonts in the executable" OFF)
+else()
+  option(USE_FONTS "Include Deja Vu and Ubuntu fonts in the executable" ON)
+  # Can download and extract latest Unbuntu font files (currently X.YY is
+  # 0.83) from:
+  # https://launchpad.net/ubuntu/+archive/primary/+files/ubuntu-font-family-sources_X.YY.orig.tar.gz
+  # TODO: automate the download and extraction of all the font and associate
+  # documentation but NOT the "sources" sub-directory contents into the
+  # ./src/fonts/ directory structure if this option is set to ON...
```

Comment:
This is not something that should be done, I don't want to see increased complexity and machinery in the build process without a _very_ good reason. We have git submodules for this and that's far enough.

Reply:
I do have a concern that some distros will be forced to drop Mudlet (or relegate it to a _lesser_ availability) because we are include material which they do not want to include in their main area.  The existing Deja Vu fonts are not a problem for Debian - who ARE a major Linux Distribution and the parent of many of the 500 or so active derivatives (consider https://en.wikipedia.org/wiki/Linux_distribution#Widely_used_distributions) but both they and Fedora have had issues with the Ubuntu fonts which have not yet been completely solved.  By not including the font files in the _source_ but only downloading them if they are wanted we could keep them happy.  Remember that distribution of the Mudlet source code will include these fonts and even if we (or a packager)  keeps them out of precompiled _binary packages_ their presence may be problematic as in holding the _source packages_ on their servers will probably count as "Propogation" as far as the Licence itself details:
> To "Propagate" a work means to do anything with it that, without permission, would make you directly or secondarily liable for infringement under applicable copyright law, except executing it on a computer or modifying a private copy. Propagation includes copying, distribution (with or without modification and with or without charging a redistribution fee), making available to the public, and in some countries other activities as well.

If I could find a Git Repo of the Ubuntu fonts I am happy to copy-paste-revise the existing sub-module code to grab the most up to date release version - but Ubuntu uses bazaar on Launchpad and whilst the binary (without the source files which requires proprietary {non-free} tools to work on those source files) are available from https://assets.ubuntu.com/v1/fad7939b-ubuntu-font-family-0.83.zip - there is not a uniform tool to download and unzip that on all platforms as far as I can tell - (unlike `git` which does have the same commands on all of them)!

Reply:
OK, it's a theoretical concern, but theory differs from practice - until we have someone ask us to change, I don't want to default to a super complicated build setup. I'd like it to be simpler, not more complicated!